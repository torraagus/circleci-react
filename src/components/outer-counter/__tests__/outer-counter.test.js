import React from 'react';
import ReactDOM from 'react-dom';
import OuterCounter from './../../outer-counter/outer-counter.component';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from 'react-test-renderer';

afterEach(cleanup);

it('should render component without crashing', () => {
    let div = document.createElement('div');
    ReactDOM.render(<OuterCounter></OuterCounter>, div);
})

it('matches snapshot', () => {
    const tree = renderer.create(<OuterCounter></OuterCounter>).toJSON();
    expect(tree).toMatchSnapshot();
})
