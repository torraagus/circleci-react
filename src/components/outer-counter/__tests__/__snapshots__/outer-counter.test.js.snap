// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="grid-counter-outer pb-3 container-fluid"
>
  <div
    className="row center-xs"
  >
    <div
      className="col-xs-12 p-0 mb-3"
    >
      <h4
        className="counterName-outer p-2 mb-0"
      >
        Outer counter
      </h4>
    </div>
  </div>
  <div
    className="row center-xs"
  >
    <div
      className="col-lg-2 col-md-4 col-sm-8 col-xs-12 p-0"
    >
      <div
        className="grid-counter-inner container-fluid"
      >
        <div
          className="row center-xs"
        >
          <div
            className="col-xs-12 p-0"
          >
            <h4
              className="counterName-inner p-2 mb-0"
            >
              Inner Counter
            </h4>
          </div>
        </div>
        <div
          className="row between-xs"
        >
          <div
            className=""
          >
            <button
              className="increaseBtnSmall"
              onClick={[Function]}
            >
              +
            </button>
          </div>
          <div
            className="counter-inner"
          >
            0
          </div>
          <div
            className=""
          >
            <button
              className="decreaseBtnSmall"
              onClick={[Function]}
            >
              -
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row center-xs"
  >
    <div
      className="col-lg-6 col-md-8 col-sm-10 col-xs-12"
    >
      <h4
        className="counter-outer"
      >
        0
      </h4>
    </div>
  </div>
  <div
    className="row center-xs"
  >
    <div
      className=""
    >
      <button
        className="increaseBtn"
        onClick={[Function]}
      >
        ADD
      </button>
    </div>
    <div
      className=""
    >
      <button
        className="decreaseBtn"
        onClick={[Function]}
      >
        REMOVE
      </button>
    </div>
  </div>
</div>
`;
